models/trigger_map.go

```diff
@@ -36,49 +36,16 @@ func (triggerMap TriggerMapModel) FirstMatchingTarget(pushBranch, prSourceBranch
 }
 
 func (triggerMap TriggerMapModel) checkDuplicatedTriggerMapItems() error {
-	triggerTypeItemMap := map[string][]TriggerMapItemModel{}
+	for i := 0; i < len(triggerMap); i++ {
```

Comment:
Maybe this could be simplified with a set? In this implementation that `triggerItem.String(false)` is called N-1 times for the last item and we could simply calculate this for every item only once. 

```
func (triggerMap TriggerMapModel) checkDuplicatedTriggerMapItems() error {
	items := make(map[string]struct{})

	for _, triggerItem := range triggerMap {
		content := triggerItem.String(false)

		_, ok := items[content]
		if ok {
			return fmt.Errorf("duplicated trigger item found (%s)", content)
		}

		items[content] = struct{}{}
	}

	return nil
}
```

Reply:
Thanks for this suggestion, updated in: https://github.com/bitrise-io/bitrise/pull/907/commits/e6cc89a0af6d40d4566f02aaa9d3230dc7fd0094